### DISCUSSION TITLE
Language and Framework Support

### DISCUSSION THREAD

**Discussion by coffee-cup** [railwayapp/nixpacks#245] [General]
Thread to chat about and request support for new languages

**Comment by coffee-cup** [comment]
Some languages that would be good to support out of the box

- [Julia](https://julialang.org/)
- [Red](https://www.red-lang.org/)
- [D](https://dlang.org/)
- [OCaml](https://ocaml.org/)

**Reply by hung222** [reply]
I want to upload django made with python 3.11 in my computer to the railways, how to get rid of the build failure and the application error port error

**Reply by coffee-cup** [reply]
Please use the #help channel on Railways [Discord](https://discord.gg/railway) for Railway specific help

**Comment by mrchypark** [comment]
[r](https://www.r-project.org/) is statistical programming language but also use as web server.

https://github.com/r-lib/actions

usefull ref.
https://eddelbuettel.github.io/r2u/

**Reply by fx2-t1d** [reply]
Agree. I'm looking for a Heroku replacement and right now the only thing that makes Railway less than perfect is that it doesn't look like it supports R-shiny.

**Comment by coffee-cup** [comment]
[Elixir](https://elixir-lang.org/) and [Phoenix](https://www.phoenixframework.org/) support

**Comment by nomeata** [comment]
What about generic nix support? I have to migrate an app off ihpcloud.com (as that shuts down), so it already has a nix-based setup, and I'd love to try railways to run it, but it's not clear to me how to integrate it best.

Or dedicated support for IHP projects would also be great :-)

**Comment by henrikbjorn** [comment]
Perl

**Comment by Yandall** [comment]
What about Nim?
